### PR TITLE
Update quickstart guide for usability

### PR DESCRIPTION
**Pull request by mitchnielsen**
Adds a few changes to the quickstart guide for usability
based on my experience running through it for the first time.

I put each change in a separate commit with a commit message for
explanation, and so we can revert/adjust any of them as needed.

I put some other notes in a [comment below](https://github.com/PrefectHQ/prefect/pull/13562#issuecomment-2130288180).

### Example

I tried to serve the docs locally, but got some errors:

```
ERROR   -  mkdocstrings: prefect.cli.kubernetes could not be found
ERROR   -  Error reading page 'api-ref/prefect/cli/kubernetes.md':
ERROR   -  Could not collect 'prefect.cli.kubernetes'

Aborted with a BuildError!
Cleaning up...
```

For now, I'll just link to the markdown file on the branch:

https://github.com/PrefectHQ/prefect/blob/update-quickstart-guide/docs/getting-started/quickstart.md

### Checklist

- ~This pull request references any related issue by including "closes `<link to issue>`"~
- ~If this pull request adds new functionality, it includes unit tests that cover the changes~
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- ~This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.~

### CODE REVIEW

File: docs/getting-started/quickstart.md [hunk 1]
```diff
@@ -22,18 +22,43 @@ Let's get started!
 
 Here's a basic script that fetches statistics about the [main Prefect GitHub repository](https://github.com/PrefectHQ/prefect).
 
-```python
-import httpx
+```python title="my_gh_workflow.py"
+import httpx   # an HTTP client library and dependency of Prefect
+from prefect import flow, task
```
Comment (billpalombi) [inline docs/getting-started/quickstart.md:27]:
```suggestion
```

**Review by billpalombi — COMMENTED** [review]
This is better for sure! We should remove that improve from the first script though.

File: docs/getting-started/quickstart.md (hunk 1 above)
Comment (mitchnielsen) [reply docs/getting-started/quickstart.md:27]: Good catch, copy pasta!

**Review by billpalombi — APPROVED** [review]
lgtm

**Review by zzstoatzz — APPROVED** [review]
🌟

**Pull request merged**